src/Umbraco.Web/Editors/MediaTypeController.cs

```diff
@@ -174,7 +176,22 @@ public MediaTypeDisplay PostSave(MediaTypeSave contentTypeSave)
         /// </summary>
         /// <param name="contentId"></param>
         [UmbracoTreeAuthorize(Constants.Trees.MediaTypes, Constants.Trees.Media)]
-        public IEnumerable<ContentTypeBasic> GetAllowedChildren(int contentId)
+        public IEnumerable<ContentTypeBasic> GetAllowedChildren(string contentId)
+        {
+            Guid idGuid = Guid.Empty;
+            int idInt;
+            if (Guid.TryParse(contentId, out idGuid)) { 
+                var entity = ApplicationContext.Services.EntityService.GetByKey(idGuid);
+                return getAllowedChildren(entity.Id);
+            } else if (int.TryParse(contentId, out idInt))
+            {
+                return getAllowedChildren(idInt);
+            }
+
+            throw new InvalidCastException("Id must be either an integer or a Guid");
+        }
+
+        private IEnumerable<ContentTypeBasic> getAllowedChildren(int contentId)

```

Comment:
Naming standards: `getAllowedChildren`  is not a valid c# method name must be GetAllowedChildren but since you already have that you could make it GetAllowedChildrenImpl or GetAllowedChildrenInternal